test/rekt/features/broker/data_plane.go

```diff
@@ -188,6 +188,38 @@ func brokerAcceptsCEVersions(ctx context.Context, t feature.T) {
 	knconf.AcceptsCEVersions(ctx, t, broker.GVR(), name)
 }
 
+func brokerAcceptsBothContentmods(ctx context.Context, t feature.T) {
+	brokerName := state.GetStringOrFail(ctx, t, "brokerName")
+
+	contenttypes := []string{
+		"application/vnd.apache.thrift.binary",
+		"application/xml",
+		"application/json",
+	}
+	for _, contenttype := range contenttypes {
+		source := feature.MakeRandomK8sName("source")
+		eventshub.Install(source,
+			eventshub.StartSenderToResource(broker.GVR(), brokerName),
+			eventshub.InputHeader("ce-specversion", "1.0"),
+			eventshub.InputHeader("ce-type", "sometype"),
+			eventshub.InputHeader("ce-source", "200.request.sender.test.knative.dev"),
+			eventshub.InputHeader("ce-id", uuid.New().String()),
+			eventshub.InputHeader("content-type", contenttype),
+			eventshub.InputBody("{}"),
+			eventshub.InputMethod("POST"),
```

Comment:
This is testing only the binary content mode because in the binary form attributes are transported using HTTP headers [1] as you're doing in this test, for example:
```
POST /someresource HTTP/1.1
Host: webhook.example.com
ce-specversion: 1.0
ce-type: com.example.someevent
ce-time: 2018-04-05T03:56:24Z
ce-id: 1234-1234-1234
ce-source: /mycontext/subcontext
    .... further attributes ...
Content-Type: application/json; charset=utf-8
Content-Length: nnnn

{
    ... application data ...
}
```

The structured mode puts all attributes in a JSON sent within the HTTP body [2]:
```

POST /myresource HTTP/1.1
Host: webhook.example.com
Content-Type: application/cloudevents+json; charset=utf-8
Content-Length: nnnn

{
    "specversion" : "1.0",
    "type" : "com.example.someevent",

    ... further attributes omitted ...

    "data" : {
        ... application data ...
    }
}
```

We have a couple of options here:
- renaming this function to `brokerAcceptsBinaryContentMode` and follow up in a separate PR to add a separate test with the structured content mode
- add the structured case in this PR with a different test

I'd go with the former but feel free to choose the option that you like the most.

[1] https://github.com/cloudevents/spec/blob/main/cloudevents/bindings/http-protocol-binding.md#31-binary-content-mode
[2] https://github.com/cloudevents/spec/blob/main/cloudevents/bindings/http-protocol-binding.md#32-structured-content-mode

Reply:
I added the structured case in this PR as a separate test. Please review the new version. Thank you.